### PR TITLE
Remove MagicValues

### PR DESCRIPTION
**Pull request by basaigh**
Feel free to push directly to this branch or make a pr with any changes.

### CODE REVIEW

**Comment by basaigh** [comment]
This does also contain a few miscellaneous fixes/cleanups, which can be removed and their own PR made.

File: src/main/java/com/github/steveice10/mc/protocol/codec/MinecraftCodecHelper.java [hunk 1]
```diff
@@ -91,7 +90,7 @@ public String readResourceLocation(ByteBuf buf) {
     }
 
     public void writeResourceLocation(ByteBuf buf, String location) {
-        this.writeString(buf, location);
+        this.writeString(buf, Identifier.formalize(location));
```
Comment (Camotoy) [inline src/main/java/com/github/steveice10/mc/protocol/codec/MinecraftCodecHelper.java]: I think this is unnecessary - this should be enforced at the constructor/setter level. Also, not all resource locations have to be formalized over the network.

File: src/main/java/com/github/steveice10/mc/protocol/data/game/inventory/ClickItemAction.java [hunk 2]
```diff
@@ -1,6 +1,31 @@
...
+    RIGHT_CLICK(1);
+
+    private final int id;
+
+    ClickItemAction(int id) {
+        this.id = id;
+    }
+
+    public int getId() {
+        return id;
```
Comment (Camotoy) [inline src/main/java/com/github/steveice10/mc/protocol/data/game/inventory/ClickItemAction.java]: You can probably replace this with the ordinal call?

File: src/main/java/com/github/steveice10/mc/protocol/data/game/inventory/CreativeGrabAction.java [hunk 3]
```diff
@@ -1,5 +1,30 @@
...
+
+    CreativeGrabAction(int id) {
+        this.id = id;
+    }
+
+    public int getId() {
+        return id;
+    }
+
+    private static Int2ObjectMap<CreativeGrabAction> VALUES = new Int2ObjectOpenHashMap<>();
```
Comment (Camotoy) [inline src/main/java/com/github/steveice10/mc/protocol/data/game/inventory/CreativeGrabAction.java]: If we know there's only one item, let's simplify the logic and not allocate a whole map.

File: src/main/java/com/github/steveice10/mc/protocol/packet/ingame/serverbound/inventory/ServerboundContainerClickPacket.java [hunk 4]
```diff
@@ -2,17 +2,8 @@
...
-import com.github.steveice10.mc.protocol.data.game.inventory.ClickItemAction;
-import com.github.steveice10.mc.protocol.data.game.inventory.ContainerAction;
-import com.github.steveice10.mc.protocol.data.game.inventory.ContainerActionType;
-import com.github.steveice10.mc.protocol.data.game.inventory.CreativeGrabAction;
-import com.github.steveice10.mc.protocol.data.game.inventory.DropItemAction;
-import com.github.steveice10.mc.protocol.data.game.inventory.FillStackAction;
-import com.github.steveice10.mc.protocol.data.game.inventory.MoveToHotbarAction;
-import com.github.steveice10.mc.protocol.data.game.inventory.ShiftClickItemAction;
-import com.github.steveice10.mc.protocol.data.game.inventory.SpreadItemAction;
+import com.github.steveice10.mc.protocol.data.game.inventory.*;
```
Comment (Camotoy) [inline src/main/java/com/github/steveice10/mc/protocol/packet/ingame/serverbound/inventory/ServerboundContainerClickPacket.java]: I think we're avoiding wildcard imports for MCProtocolLib.

**Comment by basaigh** [comment]
Thanks for the review, will get on these asap 🙂

**Review by Camotoy — APPROVED** [review]

**Pull request merged**